src/controllers/scriptController.js

```diff
@@ -97,15 +102,24 @@ const scriptController = {
   importScript: (req, res, next) => {
     const path = process.env.UPLOADPATH + req.file.filename;
 
-    scriptRepo.importScript(path)
+    scriptRepo
+      .importScript(path)
       .then((title) => {
         res.locals.title = title;
         return next();
       })
       .catch((error) => {
-        if (error.message === 'Script title already exists') {
-          // the script is already in the database, send a conflict status
-          return res.sendStatus(409);
+        // remove the uploaded file, some error occured while processing it
+        fs.unlink(path, (err) => {
+          if (err) throw err;
+        });
+        // handles errors for duplicate script and unable to find a title
+        if (error instanceof customError) {
+          return next({
```

Comment:
Could we write customError in such a way that we can just pass it into next() without having to create a new object?